src/main/java/org/broadinstitute/hellbender/transformers/DUSTReadTransformer.java

```diff
@@ -0,0 +1,230 @@
+package org.broadinstitute.hellbender.transformers;
+
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.util.Tuple;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.tools.ClipReads;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import htsjdk.samtools.SAMUtils;
+
+import java.util.*;
+
+//Masks low-complexity sequences with 'N' and base qualities with '#'
+
+public class DUSTReadTransformer implements ReadTransformer {
+    public static final long serialVersionUID = 1L;
+    private final Logger logger = LogManager.getLogger(DUSTReadTransformer.class);
+
+    @Argument(fullName="Mask PHRED score", shortName="dustPhred", optional=true)
+    public short MASK_PHRED = 15;
+
+    @Argument(fullName="Window size", shortName="dustW", optional=true)
+    public int W = 64;
+
+    @Argument(fullName="Score threshold", shortName="dustT", optional=true)
+    public float T = 20;
+
+    private static final int NUM_TRIPLETS = 64;
+
+    private class PStruct {
+        public int start = 0;
+        public int finish = 0;
+        public float score = 0;
+    }
+
+    @Override
+    public GATKRead apply(final GATKRead read) {
+
+        byte[] q = read.getBases();
+
+        //Find convert to 2-bit and assign random bases to N's
+        //TODO: reject reads with too many N's
+        Random rng = new Random();
+        rng.setSeed(0);
```

Comment:
Hate to contradict, but I think it would be a mistake to use the Utils RNG:  It will lead to thread contention and stochastic behavior.  I think it's better as it stands.